core/src/main/java/io/grpc/DecompressorRegistry.java

```diff
@@ -52,38 +51,70 @@
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1704")
 @ThreadSafe
 public final class DecompressorRegistry {
+  public static DecompressorRegistry emptyInstance() {
+    return new DecompressorRegistry();
+  }
 
-  private static final DecompressorRegistry DEFAULT_INSTANCE = new DecompressorRegistry(
-      new DecompressorInfo(new Codec.Gzip(), true),
-      new DecompressorInfo(Codec.Identity.NONE, false));
+  private static final DecompressorRegistry DEFAULT_INSTANCE =
+      emptyInstance()
+      .with(new Codec.Gzip(), true)
+      .with(Codec.Identity.NONE, false);
 
   public static DecompressorRegistry getDefaultInstance() {
```

Comment:
The default cannot be changed anymore, but anyone who is adding a custom compressor is already going to have to do something special to get this to work.  I could add a setDefault method, which would still have the same gains.  

This is already experimental so Internal is probably not needed.